src/keys.rs

```diff
@@ -248,16 +253,25 @@ impl IssuanceAuthorizingKey {
     /// Real issuance keys should be derived according to [ZIP 32].
     ///
     /// [ZIP 32]: https://zips.z.cash/zip-0032
-    pub(crate) fn random(rng: &mut impl RngCore) -> Self {
-        SpendingKey::random(rng).into()
+    pub(crate) fn random(rng: &mut impl CryptoRngCore) -> Self {
+        loop {
+            let mut bytes = [0u8; 32];
+            rng.fill_bytes(&mut bytes);
+            let schnorr_sk = schnorr::SigningKey::from_bytes(&bytes);
```

Comment:
It could be simplified by only checking it it is a `NonZeroScalar` with for instance `NonZeroScalar::from_bytes(...)`

Reply:
I don't think there is a `NonZeroScalar::from_bytes` function.